tool/teleport/common/sftp.go

```diff
@@ -89,7 +89,15 @@ func newSFTPHandler(logger *slog.Logger, req *srv.FileTransferRequest, events ch
 		}
 		// TODO(capnspacehook): reject relative paths and symlinks
 		// make filepaths consistent by ensuring all separators use backslashes
-		allowed.path = path.Clean(req.Location)
+		allowedPath, err := sftputils.ExpandHomeDir(req.Location)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+		allowedPath, err = filepath.Abs(allowedPath)
```

Comment:
IIRC the SFTP spec only handles paths with backslashes, so we should be using `path` instead of `filepath` consistently. Wouldn't this result in an error if a client supplies a Windows path with forward slashes since you use `path.Clean` below in `ensureReqIsAllowed`?